LanguageFeatures/nnbd/static_errors_A17_t03.dart

```diff
@@ -7,44 +7,21 @@
 /// Implementations that provide feedback about dead or unreachable code are
 /// encouraged to indicate that any arguments to the invocation are unreachable.
 ///
-/// @description Check that it is a warning to call a method, setter, or
+/// @description Check that it is not an error to call a method, setter, or
 /// getter on a receiver of static type Never via a null aware operator.
 /// @author sgrekhov@unipro.ru
 /// @issue 39866
 
 // Requirements=nnbd-strong
+
 void test(var x) {
   if (x is Never) {
     x?.toString();
```

Comment:
I think warnings are not reported for the unreachable code. @stereotype441 could you please confirm?

Reply:
Aha, I see what happened.  An unintentional consequence of [Flag additional code as unreachable due to types `Null` and `Never`.](https://dart-review.googlesource.com/c/sdk/+/251280) is that `if (x is Never)` no longer promotes the type of `x` to `Never`, so these errors no longer appear.

Personally I think that's a good change, so I'm inclined to leave things as they are and just update the CHANGELOG to clarify what happened.  My rationale is that promoting a variable's type to `Never` is unhelpful to users for the same reason that promoting a type to `Null` is unhelpful--it effectively hides the previous (unpromoted) type, making it impossible for the compiler to help detect errors like misspelled methods and passing the wrong number and type of arguments to methods.  The new behavior of simply marking the code unreachable, but leaving the type of the variable alone, is better.

I'll speak with some folks today to confirm whether others think this is a good idea.

Reply:
Oh, and @sgrekhov, I should mention: if you want to continue testing behaviors that result from an expression having type `Never`, you can still achieve that by having a function whose return type is `Never`.

For example instead of this:
```dart
void test(var x) {
  if (x is Never) {
    x?.toString();
    ...
```
You could do this:
```dart
Never f() => throw '';
void test(var x) {
  f()?.toString();
  ...
```

Reply:
@sgrekhov, I've discussed this behavior change with @leafpetersen and it looks like we're going to stick with the current behavior.  So I think you should feel free to either (a) land this change as is, or (b) change this test to use the `Never f() => throw '';` approach I described above, so that it continues exercising the behavior of `Never`, as intended.

Reply:
I'd recommend using `Never f() => throw '';` in order to ensure that the tests actually test the treatment of member accesses on a receiver of type `Never`.

Another thing is that when `x` is not promoted to have type `Never` then lines 27-29 in LanguageFeatures/nnbd/expression_typing_A02_t01.dart are accepted (that is, there is no error there) because `x` has type `dynamic`. Surely that's not what the test is intended to test.

Reply:
Changed the tests to use `Never foo() => throw Exception();`. Please review